pkg/workflows/sdk/runtime.go

```diff
@@ -7,9 +7,26 @@ import (
 
 var BreakErr = capabilities.ErrStopExecution
 
+type Emitter interface {
+	// Emit sends a message with the given message and labels to the configured collector.
+	//
+	// TODO(mstreet3): Emit and custmsg.Labeler should be context aware.  Update signature once
+	// WASM can support context.
+	Emit(msg string, labels map[string]any) error
+}
+
+type EmitterFunc func(msg string, labels map[string]any) error
+
+func (f EmitterFunc) Emit(msg string, labels map[string]any) error {
+	return f(msg, labels)
+}
```

Comment:
I might not be seeing this correctly, but I think with just having the function defined inside the Runtime interface we already get the benefits of decoupling it. 

Instead of having the `Emitter` interface implementing the `Emit` method that has the same signature as the `EmitterFunc` type isn't simpler to just have the 

```
type Runtime interface {
	Logger() logger.Logger
	Fetch(req FetchRequest) (FetchResponse, error)
         Emit(msg string, labels map[string]any) error
}
```

or is it because we expect the Emitter interface to grow?

Reply:
`EmitterFunc` is a convenience type/adapter so that a function can satisfy the `Emitter` interface directly.  Like `[HttpHandlerFunc](https://pkg.go.dev/net/http#HandlerFunc)` from `net/http`.  Based on @cedric-cordenier comment above though, this can be moved into the module.  It's only used so that the `beholderEmitter` function can satisfy the interface.